src/etos_client/lib/log_handler.py

```diff
@@ -106,26 +114,55 @@ def iuts(self):
             if environment.get("data", {}).get("name", "").startswith("IUT Data"):
                 yield self._iut_data(environment.get("data"))
 
+    def _download(self, name, uri, directory, spinner):
+        """Download a file and and write to disk.
+
+        :param name: Name of resulting file.
+        :type name: str
+        :param uri: URI from where the file can be downloaded.
+        :type uri: str
+        :param directory: Into which directory to write the downloaded file.
+        :type directory: str
+        :param spinner: Spinner text item.
+        :type spinner: :obj:`Spinner`
+        """
+        index = 0
+        download_name = name
+        while os.path.exists(os.path.join(directory, download_name)):
+            index += 1
+            download_name = f"{index}_{name}"
+        spinner.text = "Downloading {}".format(download_name)
+        generator = self.etos.http.wait_for_request(uri, as_json=False, stream=True)
+        try:
+            for response in generator:
+                with open(os.path.join(directory, download_name), "wb+") as report:
+                    for chunk in response:
+                        report.write(chunk)
+                break
+            return True
+        except (ConnectionError, HTTPError) as error:
+            spinner.warn("Failed in downloading '{}'.".format(download_name))
+            spinner.warn(str(error))
+            return False
+
     def download_logs(self, spinner):
         """Download all logs to report and artifact directories."""
         nbr_of_logs_downloaded = 0
         incomplete = False
 
         for name, uri in self.all_logs:
-            path = self._get_path(name)
-            spinner.text = "Downloading {}".format(name)
-            generator = self.etos.http.wait_for_request(uri, as_json=False, stream=True)
-            try:
-                for response in generator:
-                    with open(os.path.join(path, name), "wb+") as report:
-                        for chunk in response:
-                            report.write(chunk)
-                    nbr_of_logs_downloaded += 1
-                    break
-            except (ConnectionError, HTTPError) as error:
+            result = self._download(name, uri, self.report_dir, spinner)
+            if result:
+                nbr_of_logs_downloaded += 1
+            else:
+                incomplete = True
+
+        for name, uri in self.all_artifacts:
+            result = self._download(name, uri, self.artifact_dir, spinner)
+            if result:
+                nbr_of_logs_downloaded += 1
```

Comment:
```suggestion
                nbr_of_artifacts_downloaded += 1
```

Reply:
The logs variable is for printing the number of items downloaded.
Changing this to "nbr_of_artifacts_downloaded" would break the code (which is why I won't commit the suggestion) and adding another variable seems unnecessary.
I could rename it, but I don't see any great value of that since it's quite clear anyway.

Reply:
True about the brakeage, shouldn't have added it as a suggestion my bad. My reasoning regarding logs->artifacts came from the fact that you iterate over `self.all_artifacts`, but I failed miserably at seeing the entire context. Looking at it now the better name would probably be nbr_of_items_downloaded or something like that. But in retrospect it just a matter of understandability of what is being counted, and it's not that big of a deal.